### ISSUE TITLE
Request errors are silently swallowed

### ISSUE THREAD

**Issue by bo0tzz** [tomayac/local-reverse-geocoder#58]
Any errors that make it to the `request.on('error', ...)` callback (https://github.com/tomayac/local-reverse-geocoder/blob/main/index.js#L197) are swallowed silently and never thrown. Instead, the callback that's given to `geocoder.init()` is called as if initialization was successful.

Steps to reproduce:
1. Add a debug log line to the request error handler: 
```
.on('error', (err) => {
        debug(`Request error ${err}`)
        ...
```
2. Turn off network
3. Start app.js:
```
boet@Boets-Air local-reverse-geocoder % DEBUG=* npm start  
(...)
Initializing Geocoder…
(This may take a long time and will download ~300MB worth of data.)
  local-reverse-geocoder Using cities500 as override of cities database +0ms
  local-reverse-geocoder Initializing local reverse geocoder using dump directory: /Users/boet/Documents/git/local-reverse-geocoder/geonames_dump +1ms
  local-reverse-geocoder Getting GeoNames cities500 data from https://download.geonames.org/export/dump/cities500.zip (this may take a while) +1ms
  local-reverse-geocoder Getting GeoNames admin 1 codes data from https://download.geonames.org/export/dump/admin1CodesASCII.txt (this may take a while) +8ms
  local-reverse-geocoder Getting GeoNames admin 2 codes data from https://download.geonames.org/export/dump/admin2Codes.txt (this may take a while) +2ms
  local-reverse-geocoder Request error Error: getaddrinfo ENOTFOUND download.geonames.org +50ms
  local-reverse-geocoder Request error Error: getaddrinfo ENOTFOUND download.geonames.org +1ms
Geocoder initialized and ready.
Endpoints:
- http://localhost:3000/healthcheck
- http://localhost:3000/deep-healthcheck
- http://localhost:3000/geocode
Examples:
- http://localhost:3000/geocode?latitude=54.6875248&longitude=9.7617254
```

**Comment by tomayac** [comment]
Thanks for the report. Are you in a position to look into this? I'm happy to receive and review PRs (which I'll get definitely get to faster than working on a fix myself, which is "plan B").

**Comment by bo0tzz** [comment]
I managed to look into it :)

**Comment by tomayac** [comment]
Brilliant, thank you! I have just released `v0.12.5` that includes this fix.
